src/patches/dense_patch_computation.cpp

```diff
@@ -308,66 +337,85 @@ InstructionApplicationResult try_apply_instruction_direct_followup(
-            if (r.maybe_error && r.followup_instructions.empty())
-                return InstructionApplicationResult{nullptr, {instruction}};
-            if (!r.followup_instructions.empty())
-                return InstructionApplicationResult{std::make_unique<std::runtime_error>(lstk::cat(instruction,"; Followup local instructions not implemented")), {}};
+            if (!bell_init->counter.has_value())
+            {
+                auto routing_region = router.find_routing_ancilla(slice, bell_init->loc1.target, bell_init->loc1.op, bell_init->loc2.target, bell_init->loc2.op);
+                if(!routing_region) 
+                {
+                    return {std::make_unique<std::runtime_error>(lstk::cat(instruction,"; No valid route found for Bell pair creation")), {}};
+                }
+                else if (routing_region->cells.size() < 2) 
+                {
+                    return {std::make_unique<std::runtime_error>(lstk::cat(instruction,"; Shortest route cannot be used for Bell pair creation")), {}};
+                }
 
-            bell_init->counter->second++;
-        }
+                std::vector<LocalInstruction::LocalLSInstruction> local_instructions;
+                local_instructions.reserve(routing_region->cells.size());
+                std::optional<PatchId> id1; std::optional<PatchId> id2;
+                for (size_t i=0; i<routing_region->cells.size()-1; i=i+2)
+                {
+                    // Push a BellPrepare instruction with PatchID's depending on the case
+                    id1 = std::nullopt; id2 = std::nullopt;
+                    if (i==0) 
+                        id1 = bell_init->side2;
+                    if (i==routing_region->cells.size()-2)
+                        id2 = bell_init->side1;
+
+                    local_instructions.push_back({LocalInstruction::BellPrepare{id1, id2, routing_region->cells[i].cell, routing_region->cells[i+1].cell}});
+                }
+                for (size_t i=2; i<routing_region->cells.size()-1; i=i+2)
+                {
+                    // Push a complementary layer of BellMeasure instructions
+                    local_instructions.push_back({LocalInstruction::BellMeasure{routing_region->cells[i-1].cell, routing_region->cells[i].cell}});
+                }
+                // Take care of the case of an odd route
+                if ((routing_region->cells.size()%2 == 1))
+                {
+                    local_instructions.push_back({LocalInstruction::Move{
+                        routing_region->cells[routing_region->cells.size()-2].cell, 
+                        routing_region->cells[routing_region->cells.size()-1].cell,
+                        bell_init->side1
+                    }});
+                }
 
-        return {nullptr, {}};
+                bell_init->local_instructions = std::move(local_instructions);
+                bell_init->counter = std::pair<unsigned int, unsigned int>(0, 0);
```

Comment:
I'm still confused by this double counter mechanism. Would be great if the pair could be turned into a struct with named fields